transport.go

```diff
@@ -113,10 +113,29 @@ type Transport struct {
 	// If nil, context.Background() is used instead.
 	Context context.Context
 
+	// Options for cluster metadata discovery.
+	//
+	// If nil, DefaultDiscoverOptions are used.
+	DiscoverOptions *DiscoverOptions
```

Comment:
Could we have the boolean values configured on the `Transport` type directly?

Reply:
With exposed booleans we will loose backwards compatibility as default values will be set to `false`. Or we have to make ugly inversion of logic (smth like `DoNotIncludeClusterAuthorizedOperations`)

Reply:
You make a good point, let me look back into this in more details and I'll get back to you.

Reply:
Hi @achille-roussel! 

We are running into the same issue on PROD.
Is there a way to prioritise this, please?

Thanks!

Reply:
We merged a change that maybe will have a positive impact on the original issue here https://github.com/segmentio/kafka-go/pull/821

Would you be able to let me know if this does address the concern or we need to move forward with this PR?

Reply:
That's definitely solves the issue, like chopping off the legs in order not to hit your head while running 😆 

I would prefer my solution because it leaves space for future development, does not break backward compatibility and conforms with open/close principle.

Reply:
@e-chip sorry I didn't see this old PR when I raised the problem recently. See the original issue https://github.com/segmentio/kafka-go/issues/818 for what I thought the next step should be _after_ disabling the current code in order to potentially expose the ACL fetchers on the various Describe operations etc.

There's no "backward compatibility" to preserve here, because none of the existing client code was using the responses anyway, so disabling is the most immediate remedy. It also doesn't really make sense to be caching the authorized operations responses in this background metadata updater because you'd typically always want the latest information from the cluster for those.

Reply:
> I would prefer my solution because it leaves space for future development, does not break backward compatibility and conforms with open/close principle.

What I am inclined to is making sure we introduce APIs to serve known use cases; knowing how it will be employed is highly valuable to figure out what the right interface would look like. At this time it appears we haven't had a need to configure those parameters, so the best decision we can make is only driven by our experience and assumptions, therefore we are more likely to make the wrong choices.